drivers/gpio/gpio_sam.c

```diff
@@ -116,6 +107,32 @@ static int gpio_sam_config(struct device *dev, int access_op, u32_t pin,
 	return 0;
 }
 
+static int gpio_sam_config(struct device *dev, int access_op, u32_t pin,
+			   int flags)
+{
+	const struct gpio_sam_config * const cfg = DEV_CFG(dev);
+	Pio *const pio = cfg->regs;
+	int i, result;
+
+	switch (access_op) {
+	case GPIO_ACCESS_BY_PIN:
+		gpio_sam_config_pin(pio, BIT(pin), flags);
+		break;
+	case GPIO_ACCESS_BY_PORT:
+		for (i = 0; i < 32; i++) {
```

Comment:
Any reason why all the pins need to be configured even if only one of them is being requested? A comment over here can provide clarity.

Reply:
gpio_port_configure set all the pins the same way in the port, and others soc gpio drivers such as gpio_mcux.c implement thats. 

But I think it's no reason set all gpios in the same port to a same way, I opened an issue (#10305) to discuss this. I want add another param 'mask' to gpio_port_configure, only configure which pin marked in param 'mask'.

Could we discuss more details on issue #10305 about gpio_port_configure?
```
/**
 * @brief Configure all the pins the same way in the port.
 *        List out all flags on the detailed description.
 *
 * @param port Pointer to the device structure for the driver instance.
 * @param flags Flags for the port configuration. IN/OUT, interrupt ...
 * @return 0 if successful, negative errno code on failure.
 */
static inline int gpio_port_configure(struct device *port, int flags)
```